pkg/ingester/circuitbreaker_test.go

```diff
@@ -4,6 +4,7 @@ package ingester
 
 import (
 	"context"
+	"errors"
```

Comment:
Do you think it is feasible to add a test that would simulate what happens if we:
- at timestamp `ts`: schedule a cb activation with delay `3*d`
- at timestamp `ts + d`: deactivate the cb
- at timestamp `ts + 2d`: schedule another cb activation with delay `3*d`
- ensure that the cb is not active before timestamp `ts + 5d`?

Reply:
added in https://github.com/grafana/mimir/pull/9842/commits/c67e4e3402eb087e18114c4249aa5fef86b2f54e